include/Options/Options.php

```diff
@@ -490,4 +499,42 @@ public function offsetSet( $offset, $value ): void {
 	public function offsetUnset( $offset ): void {
 		$this->reset( (string) $offset );
 	}
+
+	/**
+	 * Returns the list of modified sites.
+	 * On multisite, sites are cached.
+	 * /!\ At this point, some sites may have been deleted. They are removed from `$this->modified` here.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool[]
+	 * @phpstan-return array<int, true>
+	 */
+	private function get_modified(): array {
+		if ( empty( $this->modified ) ) {
+			// Not modified.
+			return $this->modified;
+		}
+
+		// Cleanup deleted sites and cache existing ones.
+		if ( ! is_multisite() || count( $this->modified ) === 1 ) {
+			// Not multisite, or multisite with only 1 site: no need to cache or verify existence.
```

Comment:
I don't understand the thing with `count( $this->modified ) === 1`.
What do you think of:
1. I am on multisite with N>2 sites.
2. `switch_to_blog( $id )`
3. Modify the options for this site
4. Restore current blog
5. `wp_delete_site( $id )`

There are more than one site (the comment is wrong) and I modified the options of a deleted site but the code checking that is bypassed.

Reply:
Fixed in [cb39b22](https://github.com/polylang/polylang/pull/1498/commits/cb39b22a18bd30cd2122e33b836fe89b1865c649).